tools/tpm2_nvdefine.c

```diff
@@ -183,11 +183,10 @@ static void handle_default_attributes(void) {
     }
 }
 
-static tool_rc handle_no_index_specified(ESYS_CONTEXT *ectx, TPM2_NV_INDEX *chosen) {
+static tool_rc get_max_nv_index_size(ESYS_CONTEXT *ectx, UINT16 *size) {
```

Comment:
make this void, the return code isn't really used in any meaningful and consistent way. In one path, its only use is to set the default max size, in the other path it causes the tool to exit. Just make this on error set the *size to max via the define and print a warning.

Reply:
In fact the error case where it causes the tool to exit shouldn't even exist. Their should be no changes to handle_no_index_specified AFAICT.

Reply:
Ok...will make get_max_nv_index_size simpler without modifying handle_no_index_specified.